src/utilities/device_utils.h

```diff
@@ -409,6 +438,22 @@ using hypre_DeviceItem = sycl::nd_item<3>;
       assert(0); exit(1);                                                                    \
    } } while(0)
 
+#define HYPRE_CUSOLVER_CALL(call) do {                                                       \
+   cusolverStatus_t err = call;                                                              \
+   if (CUSOLVER_STATUS_SUCCESS != err) {                                                     \
+      printf("cuSOLVER ERROR (code = %d) at %s:%d\n",                                        \
+            err, __FILE__, __LINE__);                                                        \
+      hypre_assert(0); exit(1);                                                              \
+   } } while(0)
+
+#define HYPRE_ROCSOLVER_CALL(call) do {                                                      \
+   rocblas_status err = call;                                                                \
+   if (rocblas_status_success != err) {                                                      \
+      printf("rocSOLVER ERROR (code = %d, %s) at %s:%d\n",                                   \
+             err, rocblas_status_to_string(err), __FILE__, __LINE__);                        \
+      assert(0); exit(1);                                                                    \
```

Comment:
@rfalgout I'm using `exit(1)` here since that's the approach in the other macros. But I'm wondering here if we should remove that, including in the other macros, since it does not conform to the R3 policy: https://github.com/xsdk-project/xsdk-community-policies/blob/8cbe86416aff668d3ff93ea0760d8da5cb56bc65/package_policies/R3.md

Reply:
Yes, @victorapm , these should not be here.  It may be good to have an option for turning this on during debugging, as the R3 policy suggests.  Maybe through a `hypre_assert_exit` macro or something?

BTW, there are a number of other places where we call `exit` in hypre.  I've compiled a list of about 30 files that we could modify in another PR to clean this up.  Thanks!

Reply:
Thanks, Rob! Yes, maybe a `hypre_assert_exit` doing the job of `assert(0)` and the exit call. Since we are working on this, I would replace the exit call with `hypre_MPI_Abort`